pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -439,6 +444,15 @@ public CompletableFuture<Void> messageAcked(CommandAck ack) {
                 .collect(Collectors.toMap(KeyLongValue::getKey, KeyLongValue::getValue));
         }
 
+        if (subscription.isTransferring()) {
+            if (log.isDebugEnabled()) {
+                log.debug("[{}] [{}] Ignoring message acknowledgement on transferring topic, ack(s) count: {}",
+                        subscription, consumerId, ack.getMessageIdsCount());
+            }
+            return FutureUtil.failedFuture(new BrokerServiceException.TopicTransferringException(
```

Comment:
I am worried about the performance impact here, as throwing an exception needs to capture stacks. Can we complete the future without throwing this exception?  Or could you make this exception light-weight? For example disable `fillInStackTrace` and other metadata collections.